### PR TITLE
Feature/add pdf file

### PR DESCRIPTION
**Pull request by Mildred14**
Description:
I fixed comments done by my **_sensei_** and added pdf file feature!

What implements/changes?:
Scope
Migrations (t.references) 🗺 
Relationship between models(belongs_to, has_many) ❤️ 
Prawn gem💎

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,2 +1,24 @@
+require 'csv'
+require 'prawn'
 class Post < ApplicationRecord
+  belongs_to :user
```
Comment (8geonirt) [inline app/models/post.rb:5]: Add an empty line after this line, this is due to readability reasons, it is easy to read when blocks of methods are separated by empty lines.

File: app/models/post.rb [hunk 2]
```diff
@@ -1,2 +1,24 @@
...
+  def self.done
+    where(done_homework: true)
+  end
+
+  def self.pending
+    where(done_homework: false)
+  end
+
+    def self.to_csv
```
Comment (8geonirt) [inline app/models/post.rb]: Fix this indentation 🤓

File: app/pdfs/posts_pdf.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+require 'prawn'
+class PostsPdf < Prawn::Document
+    def initialize(post)
+        super()
```
Comment (8geonirt) [inline app/pdfs/posts_pdf.rb]: Fix this indentation too

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -22,8 +24,35 @@
       </tr>
     <% end %>
   </tbody>
+
 </table>
 
+<h1> Tareas Pendientes</h1>
```
Comment (8geonirt) [inline app/views/posts/index.html.erb]: The space before `Tareas` should be removed

**Review by 8geonirt — DISMISSED** [review]
Awesome job @Mildred14, I left some small comments that should be addressed, we should keep consistency in code indentation.

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,2 @@
+{
+}
```
Comment (diegosaul402) [inline .vscode/settings.json]: This file is part of your IDE, shouldn't be committed

**Review by diegosaul402 — COMMENTED** [review]
Just one comment

File: app/models/post.rb [hunk 6]
```diff
@@ -1,2 +1,24 @@
...
+  
+  belongs_to :user
+  
+  def self.done
+    where(done_homework: true)
+  end
+
+  def self.pending
+    where(done_homework: false)
+  end
```
Comment (pacocd) [inline app/models/post.rb:13]: @8geonirt @diegosaul402 should these methods be declared as scopes? (self.done and self.pending)

File: app/controllers/posts_controller.rb [hunk 7]
```diff
@@ -5,12 +5,28 @@ class PostsController < ApplicationController
...
     @posts = Post.all
+    @posts_done_homework = current_user.posts.done
+    @posts_pending_homework =  current_user.posts.pending
+    
+    respond_to do |format|
+      format.html
+      format.pdf do
+        pdf = PostsPdf.new(@posts)
+        send_data pdf.render, 
+        filename: "List_#{Date.today}.pdf",
```
Comment (diegosaul402) [inline app/controllers/posts_controller.rb]: Add indentation to the method arguments please

File: app/pdfs/posts_pdf.rb [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+     table line_items_rows
+    end
+
+    def line_items_rows
+        [["Title", "Body", "Done_homework"]]+
+        @post.map do |posts|
+            [posts.title, posts.body, "#{posts.done_homework}"]
+        end
+    end
+end
```
Comment (pacocd) [inline app/pdfs/posts_pdf.rb:19]: I see some inconsistencies in code indentation in this file

File: app/controllers/posts_controller.rb [hunk 9]
```diff
@@ -5,12 +5,28 @@ class PostsController < ApplicationController
   # GET /posts.json
   def index
     @posts = Post.all
+    @posts_done_homework = current_user.posts.done
+    @posts_pending_homework =  current_user.posts.pending
```
Comment (pacocd) [inline app/controllers/posts_controller.rb:9]: @8geonirt @diegosaul402 AFAIK these variables should be part of a separated function, what do you think?

**Review by pacocd — DISMISSED** [review]

File: app/pdfs/posts_pdf.rb [hunk 10]
```diff
@@ -0,0 +1,19 @@
+require 'prawn'
+class PostsPdf < Prawn::Document
+    def initialize(post)
+        super()
+            @post=post
+            draw_table
+    end
+
+    def draw_table
+     table line_items_rows
```
Comment (diegosaul402) [inline app/pdfs/posts_pdf.rb]: Indentation with two spaces please

File: app/pdfs/posts_pdf.rb [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+    end
+
+    def line_items_rows
+        [["Title", "Body", "Done_homework"]]+
+        @post.map do |posts|
+            [posts.title, posts.body, "#{posts.done_homework}"]
```
Comment (diegosaul402) [inline app/pdfs/posts_pdf.rb]: indentation with two spaces

**Review by diegosaul402 — CHANGES_REQUESTED** [review]
Looks good, only small issues

**Review by diegosaul402 — APPROVED** [review]
Great work! LGTM :shipit:

File: app/controllers/posts_controller.rb (hunk 7 above)
Comment (archisss) [reply app/controllers/posts_controller.rb]: iddentation for a better and readable code

File: app/pdfs/posts_pdf.rb (hunk 8 above)
Comment (archisss) [inline app/pdfs/posts_pdf.rb:19]: this file should be indented the right way

**Review by archisss — COMMENTED** [review]
you can merge

**Review by archisss — APPROVED** [review]

**Comment by edumartinez16** [comment]
Great work!

**Review by kikiondo — APPROVED** [review]

**Review by edumartinez16 — APPROVED** [review]

**Pull request merged**